components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -161,6 +189,23 @@ class P3Microphysics : public AtmosphereProcess
         // Update the atmospheric temperature and the previous temperature.
         T_atm(icol,ipack)  = PF::calculate_T_from_theta(th_atm(icol,ipack),pmid(icol,ipack));
         T_prev(icol,ipack) = T_atm(icol,ipack);
+        /*----------------------------------------------------------------------------------------------------------------------
+         *DRY-TO-WET MMRs:
+         *-----------------
+         *Since the host model (or AD) needs wet mixing ratio tendencies(state vector has wet mixing ratios),
+         *we need to convert dry mixing ratios from P3 to wet mixing ratios
+         *NOTE: water vapor mixing ratio argument in calculate_wetmmr_from_drymmr function has to be dry water vapor mixing
+         *ratio
+         *----------------------------------------------------------------------------------------------------------------------
+         */
+
+        qc(icol,ipack) = PF::calculate_wetmmr_from_drymmr(qc(icol,ipack), qv(icol,ipack));
```

Comment:
In the preamble, you had both `qc` and `qc_dry` as member variables, while here you use the same variable. I don't have a preference, but it might be easier to use the same approach in pre and post process.

Note: that's regardless of whether qc is an alias to qc_dry in memory.